From 53121d98756c5cf424944d4fdb580d85570ffa14 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 17 Jan 2015 17:55:27 -0800
Subject: [PATCH 1002/1008] MacPorts Only: Fix name of scan-view executable
 inside scan-build

http://trac.macports.org/ticket/35006

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 57af0aa8eb9a943b15475f7c71127f20dd7f6fa6)
---
 tools/scan-build/bin/scan-build | 2 ++
 1 file changed, 2 insertions(+)

diff --git llvm_release_39/tools/clang/tools/scan-build/bin/scan-build macports_release_39/tools/clang/tools/scan-build/bin/scan-build
index 3182a29767..ad951feed7 100755
--- llvm_release_39/tools/clang/tools/scan-build/bin/scan-build
+++ macports_release_39/tools/clang/tools/scan-build/bin/scan-build
@@ -1831,6 +1831,8 @@ if (defined $Options{OutputFormat}) {
       Diag "Analysis run complete.\n";
       Diag "Viewing analysis results in '$Options{OutputDir}' using scan-view.\n";
       my $ScanView = Cwd::realpath("$RealBin/scan-view");
+      if (! -x $ScanView) { $ScanView = Cwd::realpath("$RealBin/bin/scan-view"); }
+      if (! -x $ScanView) { $ScanView = Cwd::realpath("$RealBin/../scan-view/scan-view"); }
       if (! -x $ScanView) { $ScanView = "scan-view"; }
       if (! -x $ScanView) { $ScanView = Cwd::realpath("$RealBin/../../scan-view/bin/scan-view"); }
       exec $ScanView, "$Options{OutputDir}";
-- 
2.11.0

